src/Workspaces/CSharp/Portable/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -28,6 +28,11 @@ public override AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken p
                         return null;
                     }
 
+                    if (currentToken.IsPattern())
+                    {
+                        return null;
```

Comment:
why?

Reply:
Thanks. I've added a suppression on switch expression and pattern case. Then I removed this special case (on similar one below, at line 49).

Reply:
Hum, after I removed this, I get a newline before the brace when typing `_ = e is {`.
It's not showing up in tests for some reason, only in IDE at the moment.

Reply:
@CyrusNajmabadi 
Would you know a good way to test the formatting-as-you-type experience? That will help me capture and investigate behaviors.
In particular, I need to find out what generates a newline when typing `x is {`. Then I can see what prevents the same from happening when typing `if (true) { }`, and do the same.